src/transformStateWithClones.js

```diff
@@ -7,7 +7,32 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  let currentState = { ...state };
```

Comment:
[NAMING] - The variable name `currentState` is good as it clearly describes its purpose. However, for consistency and clarity, consider renaming `statesHistory` to `statesHistoryArray` or another name that indicates it's a collection of states.